crates/matrix-sdk/tests/integration/room/joined.rs

```diff
@@ -748,3 +749,24 @@ async fn subscribe_to_typing_notifications() {
     join_handle.await.unwrap();
     assert_eq!(typing_sequences.lock().unwrap().to_vec(), asserted_typing_sequences);
 }
+
+#[async_test]
+async fn get_power_level_for_user() {
+    let (client, server) = logged_in_client_with_server().await;
+
+    mock_sync(&server, &*test_json::DEFAULT_SYNC_SUMMARY, None).await;
+
+    let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
+    let _response = client.sync_once(sync_settings).await.unwrap();
+    let room = client.get_room(&DEFAULT_TEST_ROOM_ID).unwrap();
+
+    let role_admin =
+        room.get_suggested_user_role(UserId::parse("@example:localhost").unwrap()).await.unwrap();
+    assert_eq!(role_admin, RoomMemberRole::Administrator);
```

Comment:
I know it's indirectly tested, but can you also test directly `get_user_power_level` please?

Reply:
Done in https://github.com/matrix-org/matrix-rust-sdk/pull/3170/commits/600cd173aed50560b4553907295f3533780b1d55